lib/msf/core/exploit/remote/ldap_server.rb

```diff
@@ -0,0 +1,80 @@
+# -*- coding: binary -*-
+
+module Msf
+  ###
+  #
+  # This module provides methods for exploiting an LDAP client by acting
+  # as an LDAP server.
+  #
+  ###
+  module Exploit::Remote::LdapServer
```

Comment:
@timwr Would you be up for collaborating on this reusable LDAP server mixin? I started on this as well over in #15958 for testing. I think there's a good opportunity to further utilize the existing net-ldap library we're already including to fully parse out all of the requests. By extending the socket object with `Net::BER::BERParser` we can read in fully parsed LDAP PDUs and not have to deal with the ASN1 parsing ourselves.

Reply:
Sure that more sounds sensible. The ldap stuff looks straightforward but I'm not sure about the best way to do java deserialization. Perhaps we could harness ysoserial for everything (including jython).